### PR TITLE
feat(client/linux): generate Debian package

### PR DESCRIPTION
**Pull request by jyyi1**
Adds support for generating a Debian package (.deb) for Linux Outline Client.

This allows users to install Outline Client via `sudo apt install Outline-Client.deb`. The package [installs the client to `/opt/Outline`](https://github.com/electron-userland/electron-builder/blob/v28.0.0/packages/app-builder-lib/src/targets/LinuxTargetHelper.ts#L72), it also creates a launch shortcut.

Uninstalling the package with `sudo apt remove/purge outline-client` removes everything.

This PR adds two "name"s as well:

-  `"extraMetadata"."name"`:  Sets the apt package name (visible via `apt list | grep outline-client`).
-  `"linux"."executableName"`:  Sets the executable file name in `/opt/Outline/`. The first letter is capitalized ("Outline") to avoid icon conflicts with Ubuntu's default "outline" icon in launch shortcut. 

### TODO
- [ ] Auto-update will be disabled in a separate PR
- [ ] **Service-less architecture** will be implemented in a separate PR

### CODE REVIEW

File: client/electron/electron-builder.json [hunk 1]
```diff
@@ -15,21 +16,35 @@
     "output/client/electron",
     "!output/client/electron/build"
   ],
+
+  "deb": {
+    "depends": [
+      "gconf2", "gconf-service", "libnotify4", "libappindicator1", "libxtst6", "libnss3",
```
Comment (jyyi1) [inline client/electron/electron-builder.json:22]: These are electron required dependencies: https://www.electron.build/app-builder-lib.interface.deboptions#depends

Comment (fortuna) [reply client/electron/electron-builder.json:22]: Can you add them as comments to the file? Ugh, it's JSON...

File: client/electron/electron-builder.json [hunk 2]
```diff
@@ -15,21 +16,35 @@
...
-    "target": {
+    "maintainer": "The Outline Authors",
+    "target": [{
       "arch": [
         "x64"
       ],
       "target": "AppImage"
-    }
+    }, {
+      "arch": "x64",
```
[the remark this answers is not in the document]
Comment (jyyi1) [reply client/electron/electron-builder.json:43]: We can, we also need to update Go compiler arguments for `tun2socks`. (And I don't have a ARM based machine to test) Could we add this later?

File: client/electron/electron-builder.json [hunk 3]
```diff
@@ -15,21 +16,35 @@
...
     "category": "Network",
+    "executableName": "Outline",
     "files": [
       "client/electron/linux_proxy_controller/dist",
       "client/electron/icons/png",
       "client/output/build/linux"
     ],
     "icon": "client/electron/icons/png",
-    "target": {
+    "maintainer": "The Outline Authors",
```
[the remark this answers is not in the document]
Comment (jyyi1) [reply client/electron/electron-builder.json]: Updated to `Jigsaw LLC`

File: client/electron/electron-builder.json (hunk 1 above)
Comment (jyyi1) [reply client/electron/electron-builder.json:22]: Right, JSON is bad. How about adding the field `"_depends_comment": "there are electron required deps"`?

**Review by fortuna — APPROVED** [review]

**Pull request merged**